opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -164,6 +168,22 @@ namespace Opm
                 solver->step(timer.currentStepLength(), state, well_state);
             }
 
+            // update the derived geology (transmissibilities, pore volumes, etc) if the
+            // has geology changed for the next report step
+            const int nextTimeStepIdx = timer.currentStepNum() + 1;
+            if (nextTimeStepIdx < timer.numSteps()
+                && events.hasEvent(ScheduleEvents::GEO_MODIFIER, nextTimeStepIdx)) {
+                // bring the contents of the keywords to the current state of the SCHEDULE
+                // section
+                //
+                // TODO (?): handle the parallel case (maybe this works out of the box)
+                ScheduleConstPtr schedule = eclipse_state_->getSchedule();
+                DeckConstPtr miniDeck = schedule->getModifierDeck(timer.currentStepNum() + 1);
+                eclipse_state_->applyModifierDeck(miniDeck);
+                const bool use_local_perm = param_.getDefault("use_local_perm", true);
```

Comment:
We should avoid reading the parameters after the simulation has started, rather make this a class member.